### PR TITLE
Fix _expand_archive()

### PR DESCRIPTION
**Pull request by gabriel-samfira**
In cases where System.IO.Compression.ZipFile is available on the
windows minion, if the destination folder exists, it will cause
an error.

Where this component is not available, if the target folder exists
and there are conflicting files inside it with what's inside the
archive, it will error out.

In both cases, ensuring that the destination folder does not exist
will fix the issue.

If the destination folder is the root folder or is within the
%WINDOWS% folder, we do not remove it.

### CODE REVIEW

File: coriolis/osmorphing/windows.py [hunk 1]
```diff
@@ -146,9 +147,21 @@ def _dismount_disk_image(self, path):
...
+    def _expand_archive(self, path, destination, overwrite=True):
         LOG.info("Expanding archive \"%(path)s\" in \"%(destination)s\"",
                  {"path": path, "destination": destination})
+
+        if self._conn.test_path(destination):
+            LOG.info("Destination folder %s already exists" % destination)
+            if overwrite:
+                if destination.endswith(":\\") or ":\\Windows" in destination:
+                    LOG.warn(
+                        "Not removing target directory, as it is either the "
```
Comment (aznashwan) [inline coriolis/osmorphing/windows.py]: Do we run the risk of it asking for overwrite confirmation and hanging if we ignore this here?
If so, I suggest we just error out and impose that the plugins not directly unarchive shit in C:\

Comment (gabriel-samfira) [reply coriolis/osmorphing/windows.py]: the expand archive commands are non-interactive.

Comment (aznashwan) [reply coriolis/osmorphing/windows.py]: Cool. LGTM then.

**Comment by aznashwan** [comment]
LGTM. Merging.

**Pull request merged**